src/qutip_tensorflow/core/data/dense_tensor.py

```diff
@@ -0,0 +1,85 @@
+import qutip as qt
+import numpy as np
+import tensorflow as tf
+import qutip.core.data as data
```

Comment:
You should protect this name with an underscore or something - you frequently use `data` as a variable, so at several points during the code, it's not at all clear what the identifier `data` refers to.